docs/ai/shade-agents/introduction.md

```diff
@@ -0,0 +1,94 @@
+
+If the attestation quote is valid and the code hash matches the expected code hash of the worker agent (defined during agent contract deployment), the worker agent's account is approved as a valid worker agent.
+
+Once registered, the worker agent can call the `sign_tx` function on the agent contract, enabling it to sign transactions on behalf of the Shade Agent. The Shade Agent utilizes [chain signatures](../../chain-abstraction/chain-signatures.md) for decentralized key management, allowing it to hold assets and sign transactions on nearly any chain.
+
+Using the correct Docker image, `anyone` can deploy a worker agent to a TEE, register as a valid worker, and access the sign_tx function to control the same Shade Agent accounts. This means the accounts are never lost, and the service is `always live`. Worker agents are designed to be `stateless` to facilitate this functionality.
+
+<details>
+
+<summary>What is a TEE?</summary>
+
+A trusted execution environment is a secure area of a CPU that runs code in an isolated and protected way. This means we know the expected code is running and its execution is not exposed outside of the enclave. TEEs produce attestations to prove that the code is running within a TEE and that it's running the specified code.
+
+</details>
+
+---
+
+## Agent Contract Functions
+
+The Agent Contract has three main functions:
+
+#### Approve Code Hash
+
+After contract deployment, a `code hash` is set to ensure that only worker agents running the correct worker code can register.
+
+Developers or protocols may need to modify the code running inside worker agents. When the contract is first deployed, it's initialized with an `owner` account Id. Only the owner can call the approve code hash function to update the worker agent code.
+
+:::tip
+Upgradeability can be designed to meet the agent's requirements. Common upgrade methods include approving a new code hash through DAO voting or implementing a grace period or cooldown, allowing users to withdraw funds if they're uncomfortable with the incoming code hash for the new worker agent.
+:::
+
+#### Register Worker
+
+The register worker function verifies that the worker agent is running in a TEE and executing the expected code. Upon successful verification, the worker agent's account Id becomes registered and gains access to `gated functions` (sign_tx).
+
+#### Sign Tx
+
+The `sign_tx` function serves as the gateway to access the Shade Agent's multichain accounts and sign transactions. It employs `method access control` so that only valid worker agent account Ids are able to use this function. Developers should not sign transactions from the worker agents themselves (except for functions on the agent contract) as these accounts will be lost if the worker agent goes down. The magic of Shade Agents is that the multichain accounts are tied to the agent contract, which are persistent and accessible by any instance of a valid worker agent.
```

Comment:
```suggestion
The `sign_tx` function serves as the gateway to access the Shade Agent's multichain accounts and sign transactions. It employs `method access control` so that only the registered `workers` are able to use this function. 

:::tip
Developers should not sign transactions from the worker agents themselves (except for functions on the agent contract) as these accounts will be lost if the worker agent goes down.

The magic of Shade Agents is that the multichain accounts are tied to the agent contract, which are persistent and accessible by any instance of a valid worker agent.
:::
```